lib/src/signed_video_h26x_auth.c

```diff
@@ -720,6 +721,41 @@ prepare_for_validation(signed_video_t *self)
   return status;
 }
 
+// If public_key is not received then try to decode all recurrent tags.
+static bool
+is_recurrent_data_decoded(signed_video_t *self)
+{
+  gop_state_t *gop_state = &(self->gop_state);
+  gop_info_detected_t *gop_info_detected = &(self->gop_info_detected);
+  signed_video_latest_validation_t *latest = self->latest_validation;
+  h26x_nalu_list_t *nalu_list = self->nalu_list;
+
+  if (!self->has_public_key && gop_state->signing_present) {
+    bool recurrent_data_decoded = false;
+    h26x_nalu_list_item_t *item = nalu_list->first_item;
+
+    while (item && !recurrent_data_decoded) {
+      if (item->nalu && item->nalu->is_gop_sei && item->validation_status == 'P') {
+        const uint8_t *tlv_data = item->nalu->tlv_data;
+        size_t tlv_size = item->nalu->tlv_size;
+        recurrent_data_decoded =
+            tlv_find_and_decode_recurrent_tags(self, tlv_data, tlv_size) == SVI_OK;
+        if (!recurrent_data_decoded) {
+          DEBUG_LOG("Failed to find and decode recurrent tags");
+        }
+      }
+      item = item->next;
+    }
+    if (!recurrent_data_decoded) {
+      DEBUG_LOG("Public key missing");
+      gop_state_reset(gop_state, gop_info_detected);
```

Comment:
I think we should add a TODO to investigate if we need to take any special actions when moving to the next GOP. Recall the test we did the other day.